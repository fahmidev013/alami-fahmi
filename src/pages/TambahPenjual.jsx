import React from "react";
import PropTypes from "prop-types";
import { useHistory } from "react-router-dom";
import axios from "axios";
import ModalComponent from "../components/ModalComponent";

const TambahPenjual = () => {
  const history = useHistory();
  const [modalUp, setModalUp] = React.useState(false);
  const [modalContent, setmodalContent] = React.useState({
    isSuccess: false,
    title: "",
    content: "",
  });

  const onFinish = (event) => {
    event.preventDefault();
    axios
      .post(`addSeller`, {
        nama: event.target.nama.value,
        kota: event.target.kota.value,
      })
      .then((response) => response.data)
      .then((result) => {
        setmodalContent({
          isSuccess: true,
          title: "Sukses Menambah Penjual",
          content: "Data berhasil ditambah",
        });
      })
      .catch(function (error) {
        console.log("Failed Addseller " + error);
        setmodalContent({
          isSuccess: false,
          title: `Gagal Tambah Penjual`,
          content: error.response.data.message,
        });
      })
      .finally(() => setModalUp(true));
  };

  return (
    <>
      <div className="site-layout-content">
        <form onSubmit={onFinish}>
          <ul className="wrapper">
            <li className="form-row">
              <label htmlFor="nama">Nama :</label>
              <input type="text" id="nama" />
            </li>
            <li className="form-row">
              <label htmlFor="kota">Kota :</label>
              <input type="text" id="kota" />
            </li>
            <li className="form-row">
              <label htmlFor=""></label>
              <button type="submit">Simpan</button>
            </li>
          </ul>
        </form>
      </div>
      <ModalComponent
        type={modalContent}
        visible={modalUp}
        onConfirm={() => {
          setModalUp(false);
          history.push("/");
        }}
      />
    </>
  );
};

TambahPenjual.propTypes = {};

export default TambahPenjual;
