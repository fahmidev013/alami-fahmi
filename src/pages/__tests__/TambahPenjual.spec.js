import React from 'react';
import TambahPenjual from '../TambahPenjual';
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Tambah Penjual Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<TambahPenjual />);
  });

  it("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});

