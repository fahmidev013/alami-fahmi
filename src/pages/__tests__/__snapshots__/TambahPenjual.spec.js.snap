// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tambah Penjual Component should render correctly 1`] = `
<Fragment>
  <div
    className="site-layout-content"
  >
    <form
      onSubmit={[Function]}
    >
      <ul
        className="wrapper"
      >
        <li
          className="form-row"
        >
          <label
            htmlFor="nama"
          >
            Nama :
          </label>
          <input
            id="nama"
            type="text"
          />
        </li>
        <li
          className="form-row"
        >
          <label
            htmlFor="kota"
          >
            Kota :
          </label>
          <input
            id="kota"
            type="text"
          />
        </li>
        <li
          className="form-row"
        >
          <label
            htmlFor=""
          />
          <button
            type="submit"
          >
            Simpan
          </button>
        </li>
      </ul>
    </form>
  </div>
  <ModalComponent
    onConfirm={[Function]}
    type={
      Object {
        "content": "",
        "isSuccess": false,
        "title": "",
      }
    }
    visible={false}
  />
</Fragment>
`;
